Azure/azure-cli
# [Keyvault] Fix #18319 & #21555: `az keyvault list-deleted`: List all deleted resources if no specified resource type

**Description**

Fix #18319 #21555 
`az keyvault list-deleted` without `--resource-type` should list all deleted resources according to the help message:
```
Arguments
    --resource-type    : When --resource-type is not present the command will list all deleted
                         Vaults and HSMs. Possible values for --resource-type are vault and hsm.
```
But it only list deleted vaults for now.

**Testing Guide**

Prepare deleted vault & deleted hsm
`az keyvault list-deleted`
`az keyvault list-deleted --resource-type vault`
`az keyvault list-deleted --resource-type hsm`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
